### DISCUSSION TITLE
Show self used pv energy (in kWh) on energy dashboard

### DISCUSSION THREAD

**Discussion by tremor69** [home-assistant/frontend#15131] [Dashboard]
The energy dashboard is great for monitoring the energy flows between grid/pv/home. But one metric I am missing from the dashboard is the amount of PV kWh that have been used locally (e.g. the yellow parts of the energy chart). There is a widget that shows the percentage of pv output used locally, but not the absolute amount. I use it to estimate my savings on energy cost from the grid. The number is already available by calculation, but maybe there is an easy way to add it to the pv self usage chart without too much work.

I have neither found an issue or previous discussion on the topic and thought I might ask here. I am a developer and would be willing to look into the implementation if this feature is considered worthwhile.

**Comment by Gitsarry** [comment]
I would suggest the Energy dashboard should have one additional line above the "Solar total" line:

Solar energy self consumed   -> [kWh]    ~negative cost: <value * grid consumption price>~

e.g. "self consumed   10 kWh"

Edit: Removed the negative cost, as that is non-trivial and off-topic from the OP.

**Reply by markkulusa** [reply]
Still wishing for this to be implemented, shouldn't be a big thing to make tariff for "saved" money in the dashboard...

**Reply by Pliggen** [reply]
I had fixed pricing on my electricity until last month and that was easy to calculate in a sensor in Home Assistant. Now I have an average pricing set at end of month based on hourly pricing throu the month and I cant figure out how to make the calculation on saved value on self-consumed solar generated energy.
So, 
how do we notice them that we want them to implement this?

**Reply by Gitsarry** [reply]
Hi, I just removed the negative cost from my [comment](https://github.com/home-assistant/frontend/discussions/15131#discussioncomment-5724151) as I revognized that is non-trivial and off-topic from the OP.

For the non-trivial cost-saving request, please see the [discussion here](https://github.com/home-assistant/frontend/discussions/16308)

**Comment by AndersHoglund** [comment]
Yes please, self consumption in both absolute and relative numbers. Also published as a sensor, making it reusable for further computations.

**Reply by Adamou02** [reply]
Is it published as a sensor ? Or you ask for a request ?

**Comment by Pliggen** [comment]
Please look in to this and see if you could implement it in a future update, would be really useful.

**Comment by Gitsarry** [comment]
> I am a developer and would be willing to look into the implementation if this feature is considered worthwhile.

I think you can see from the comments here that the feature is worthwile. I would suggest that you create a pull request where you reference this issue. As you have written, "The number is already available by calculation"..

**Comment by Adamou02** [comment]
+1 for this. Very useful

**Comment by Pliggen** [comment]
It seems like the data is calculated somewhere in Home Assistant but how
can we use it as our own sensor?

Den fre 4 apr. 2025 16:30Adam Bouhrara ***@***.***> skrev:

**Reply by Adamou02** [reply]
I don't know. Maybe we can't do it actually
